// es-queries/places.js
var placesQueries = {
    getAll: function(q){
        return {
            index: 'places',
            body: {
                size: 200,
                query: {
                    filtered: {
                        filter: {
                            geo_distance: {
                                distance: (q.distance / 1000) + 'km',
                                location: [+q.lat, +q.lon]
                            }
                        },
                        query: {
                            bool : {
                                must : {
                                    terms : { "tags" : q.tags, minimum_should_match : 1}
                                }
                            }
                        }
                    }
                }
            }
        }
    },
    getOne: function(id){
        return {
            index: 'places',
            body: {
                query: {
                    "match": {
                       "_id": id
                    }
                }
            }
        }
    }
};

module.exports = placesQueries;
